src/metabase/sync/events/sync_database.clj

```diff
@@ -0,0 +1,31 @@
+(ns metabase.sync.events.sync-database
+  (:require
+   [metabase.events :as events]
+   [metabase.sync.concurrent :as sync.concurrent]
+   [metabase.sync.sync :as sync]
+   [metabase.sync.sync-metadata :as sync-metadata]
+   [metabase.util :as u]
+   [metabase.util.log :as log]
+   [methodical.core :as methodical]))
+
+(derive ::event :metabase/event)
+(derive :event/database-create ::event)
+
+(methodical/defmethod events/publish-event! ::event
+  "When a new Database is created, kick off a sync process for it in a different thread."
+  [topic {database :object :as _event}]
+  ;; try/catch here to prevent individual topic processing exceptions from bubbling up.  better to handle them here.
+  (try
+    (when database
+      ;; just kick off a sync on another thread
+      (sync.concurrent/submit-task!
```

Comment:
Here's another free bugfix. The old version of this was using `future` instead of using the sync threadpool.